parser/src/parse.rs

```diff
@@ -0,0 +1,242 @@
+use std::num::ParseIntError;
+
+/// Error type of [unescape](unescape).
+#[derive(Debug, PartialEq)]
+pub enum ParseError {
+    // #[error("invalid escape {escape} at {index} in {string}")]
```

Comment:
The original library used [thiserror](https://docs.rs/snailquote/latest/src/snailquote/lib.rs.html#12) - I kept it for now, but happy to adjust error handling if you prefer to do it differently.